### PR TITLE
Fix saving connections in the nmcli editor [bg/cli-editor-save-rh1546805]

### PR DESCRIPTION
**Pull request by bengal**
https://bugzilla.redhat.com/show_bug.cgi?id=1546805

### CODE REVIEW

**Comment by thom311** [comment]
can you add a `Fixes` comment for 474a0dbfbeeda7504d6599abe4adf0ddf18bab1e?

File: clients/cli/connections.c [hunk 1]
```diff
@@ -7764,10 +7772,20 @@ editor_menu_main (NmCli *nmc, NMConnection *connection, const char *connection_t
 					update_connection (persistent, rem_con, update_connection_editor_cb, NULL);
 				}
 
-				//FIXME: add also a timeout for cases the callback is not called
+				nmc_editor_connection_changed = FALSE;
+				handler_id = g_signal_connect (rem_con,
+				                               NM_CONNECTION_CHANGED,
+				                               G_CALLBACK (editor_connection_changed_cb),
+				                               NULL);
```
Comment (thom311) [inline clients/cli/connections.c]: can we do this without globals? Just pass a reference to a variable on the stack as user-data.

**Comment by vhumpa** [comment]
rebuild

**Comment by vhumpa** [comment]
rebuild

**Comment by vhumpa** [comment]
rebuild

**Comment by vhumpa** [comment]
The beaker systems we use to test veth are currently down I understand. So I've changed those systems to different VMs temporalily to have this working until at least the gitlab ci is done (which being 'veth' can run on openstack for this code repo).

**Comment by vhumpa** [comment]
rebuild

File: clients/cli/connections.c [hunk 2]
```diff
@@ -7764,10 +7771,20 @@ editor_menu_main (NmCli *nmc, NMConnection *connection, const char *connection_t
...
+				connection_changed = FALSE;
+				handler_id = g_signal_connect (rem_con,
+				                               NM_CONNECTION_CHANGED,
+				                               G_CALLBACK (editor_connection_changed_cb),
+				                               &connection_changed);
+
 				while (!nmc_editor_cb_called)
 					g_main_context_iteration (NULL, TRUE);
 
+				while (!connection_changed)
```
Comment (thom311) [inline clients/cli/connections.c]: if the signal for some reason does not come, this loops indefinitely. I think we need a timeout here, maybe 5 seconds?

The rest lgtm.

**Comment by bengal** [comment]
Merged as 8af4f76638609549d151b449ee7c7fbc716e5d1d and backported to nm-1-14.

**Pull request closed without merging**